Firestore/Example/Tests/Local/FSTLevelDBMigrationsTests.mm

```diff
@@ -70,6 +72,21 @@ - (void)testSetsVersionNumber {
   XCTAssertGreaterThan(actual, 0, @"Expected to migrate to a schema version > 0");
 }
 
+- (void)testCountsQueries {
+  NSUInteger expected = 50;
+  FSTWriteGroup *group = [FSTWriteGroup groupWithAction:@"Setup"];
+  for (int i = 0; i < expected; i++) {
+    std::string key = [FSTLevelDBTargetKey keyWithTargetID:i];
+    [group setData:"dummy" forKey:key];
+  }
+  Status status = [group writeToDB:_db];
+  XCTAssertTrue(status.ok(), @"Failed to write targets");
+
+  [FSTLevelDBMigrations runMigrationsOnDB:_db];
+  FSTPBTargetGlobal *metadata = [FSTLevelDBQueryCache readTargetMetadataFromDB:_db];
+  XCTAssertEqual(expected, metadata.targetCount, @"Failed to count all of the targets we added");
```

Comment:
I will add a key to test this, but it appears that the way we construct keys includes a table name and terminator in the prefix. I'm fairly certain that means as a general rule we're ok using a full tablename as a prefix for iteration bounds. There actually already exists an adversarial key, the target_global table/key (vs a table name of 'target'), but it is not explicit in the test.

Reply:
Right: I added the explicit terminators so that we couldn't be tricked by "food" when searching in the table "foo". However, having been bitten by this by failing to use the routines correctly I'm paranoid.